pkg/kubelet/images/image_gc_manager.go

```diff
@@ -182,19 +182,35 @@ func (im *realImageGCManager) Start() {
 		} else {
 			im.initialized = true
 		}
-	}, 5*time.Minute, wait.NeverStop)
+	}
 
-	// Start a goroutine periodically updates image cache.
-	// TODO(random-liu): Merge this with the previous loop.
-	go wait.Until(func() {
+	updateImages := func() {
 		images, err := im.runtime.ListImages()
 		if err != nil {
 			klog.Warningf("[imageGCManager] Failed to update image list: %v", err)
 		} else {
 			im.imageCache.set(images)
 		}
-	}, 30*time.Second, wait.NeverStop)
+	}
 
+	// Should we want/need to for testing, we could make this `clock` variable
+	// part of the `ImageGCManager` struct, which would make mocking
+	// trivial.
+	clock := clock.RealClock{}
+
+	imageMonitorTicker := clock.NewTicker(5 * time.Minute)
+	defer imageMonitorTicker.Stop()
+	imageUpdaterTicker := clock.NewTicker(30 * time.Second)
+	defer imageUpdaterTicker.Stop()
+
+	for {
```

Comment:
Hmm, isn't the difference here that in the current version `Start()` returns immediately but in your version `Start()` never returns?

Reply:
Fantastic point :) Fixing!